src/managers/input/InputManager.cpp

```diff
@@ -1036,6 +1043,50 @@ void CInputManager::newTouchDevice(wlr_input_device* pDevice) {
     }, PNEWDEV, "TouchDevice");
 }
 
+void CInputManager::setTouchDeviceConfigs() {
+    // The rotation matrices.
+    // The third row is always 0 0 1 and is not expected by `libinput_device_config_calibration_set_matrix`
+    const float MATRICES[4][6] = {
+       {
+        1, 0, 0,
+        0, 1, 0
+       },
+       {
+        0, -1, 1,
+        1, 0, 0
+       },
+       {
+        -1, 0, 1,
+        0, -1, 1
+       },
+       {
+        0, 1, 0,
+        -1, 0, 1
+       }
+    };
+    const int NB_MATRICES = 4;
+    for (auto& m : m_lTouchDevices) {
+        const auto PTOUCHDEV = &m;
+
+        auto devname = PTOUCHDEV->name;
+        transform(devname.begin(), devname.end(), devname.begin(), ::tolower);
+
+        const auto HASCONFIG = g_pConfigManager->deviceConfigExists(devname);
+
+        if (HASCONFIG)
+            Debug::log(LOG, "Touch Screen %s has config", devname.c_str());
+        else
+            Debug::log(LOG, "Touch Screen %s has no config", devname.c_str());
+
+        if (wlr_input_device_is_libinput(m.pWlrDevice)) {
+            const auto LIBINPUTDEV = (libinput_device*)wlr_libinput_get_device_handle(m.pWlrDevice);
+
+            const int ROTATION = ((HASCONFIG ? g_pConfigManager->getDeviceInt(devname, "td_rotation") : g_pConfigManager->getInt("input:touchdevice:td_rotation")) % NB_MATRICES + NB_MATRICES) % NB_MATRICES;
```

Comment:
oh god jesus christ what in the fuck

as above - can't you just detect the output's transform??

Reply:
I'm not particularly proud of that. I didn't find a way to enforce a bound on the value set in the config, so this is the next best solution to avoid out of bound. If you have a better alternative I'll gladly implement it over this.
I didn't see any bounds check for the value of `TRANSFORM` (in `monitor=NAME,transform,TRANSFORM`), so using the value from the config won't solve this issue.

Reply:
it's called `std::clamp`

Reply:
I need to go back to C++ school